apps/dcos_dns/src/dcos_dns_mesos.erl

```diff
@@ -0,0 +1,384 @@
+-module(dcos_dns_mesos).
+-behavior(gen_server).
+
+-include("dcos_dns.hrl").
+-include_lib("dns/include/dns.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+%% API
+-export([
+    start_link/0,
+    dns_records/2,
+    push_zone/2
+]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2,
+    handle_info/2, terminate/2, code_change/3]).
+
+-type task() :: dcos_net_mesos_state:task().
+-type task_id() :: dcos_net_mesos_state:task_id().
+
+-define(DCOS_DNS_TTL, 5).
+-define(IS_RUNNING(TS), not is_boolean(TS)).
```

Comment:
IS_RUNNING logic is very cryptic. I have a feeling that it could be made easier to understand